### PR TITLE
[Clean Up] - NodeStageVolume, NodeUnStageVolume, Adding golangci lint to GHA, and various small cleans

### PR DESCRIPTION
**Pull request by komer3**
### General:

* [ ] Have you removed all sensitive information, including but not limited to access keys and passwords?
* [ ] Have you checked to ensure there aren't other open or closed [Pull Requests](../../pulls) for the same bug/feature/question?

### Pull Request Guidelines:

1. [ ] Does your submission pass tests?
1. [ ] Have you added tests? 
1. [ ] Are you addressing a single feature in this PR? 
1. [ ] Are your commits atomic, addressing one change per commit?
1. [ ] Are you following the conventions of the language? 
1. [ ] Have you saved your large formatting changes for a different PR, so we can focus on your work?
1. [ ] Have you explained your rationale for why this feature is needed? 
1. [ ] Have you linked your PR to an [open issue](https://blog.github.com/2013-05-14-closing-issues-via-pull-requests/)

### CODE REVIEW

File: pkg/common/idhelpers.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ type (
 	}
 )
 
-const linodeBSLabelLength = 32
+// TODO: Rename this variable
+const linodeStorageLabelLength = 32
```
Comment (AshleyDumaine) [inline pkg/common/idhelpers.go:24]: might want to change this and other variables from storage to volume?

Comment (komer3) [reply pkg/common/idhelpers.go:24]: Oh yup linodeVolumeLabelLenght is more fitting!

**Review by shanduur — APPROVED** [review]

**Comment by eljohnson92** [comment]
it looks like you need remove the `./mock` directory from code coverage reporting here

File: devbox.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "packages": [
-    "go_1_22@latest",
```
Comment (eljohnson92) [inline devbox.json:3]: I think we should keep this pinned to 1.22 if we can as upgrades past 1.22 will likely need some more intentional work to do

File: internal/driver/luks.go [hunk 3]
```diff
@@ -122,20 +132,23 @@ func luksFormat(ctx LuksContext, source string) error {
...
-	if err != nil {
-		return fmt.Errorf("failed to get stdin pipe for cryptsetup, got err: %w", err)
-	}
+	// Set the encryption key as stdin
+	cmd.SetStdin(strings.NewReader(ctx.EncryptionKey))
 
-	if _, err := io.WriteString(stdin, ctx.EncryptionKey); err != nil {
-		return fmt.Errorf("failed to write to stdin pipe for cryptsetup, got err: %w, closing pipe: %w", err, stdin.Close())
-	}
+	// stdin, err := cmd.StdinPipe()
```
Comment (eljohnson92) [inline internal/driver/luks.go:140]: we should remove commented-out code here or describe why it needs to stay commented out, a comment describing what functionality we might want to add in later is likely a better approach

Comment (komer3) [reply internal/driver/luks.go:140]: I'm leaving it here until Luks E2E tests are merged in. Its to make sure the new code works and didn't break luks functionality.

File: internal/driver/nodeserver.go [hunk 4]
```diff
@@ -236,53 +234,16 @@ func (ns *LinodeNodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeSt
 		return &csi.NodeStageVolumeResponse{}, nil
 	}
 
-	// VolumeMode=block
+	// Check if the volume mode is set to 'Block'
 	// Do nothing else with the mount point for stage
-	if blk := volumeCapability.GetBlock(); blk != nil {
+	if blk := req.VolumeCapability.GetBlock(); blk != nil {
```
Comment (bcm820) [inline internal/driver/nodeserver.go:239]: Do we need a `nil` check on `req.VolumeCapability` here since the original implementation had one?

Comment (komer3) [reply internal/driver/nodeserver.go:239]: Yup. I'm trying to keep the original code the same and not make any changes unless absolutely needed

Comment (bcm820) [reply internal/driver/nodeserver.go:239]: Oh, I was not referring to the return value of `GetBlock`, but `req.VolumeCapability` itself. I just noticed that there was a nil check on it before but I don't see it being done here.

**Review by bcm820 — APPROVED** [review]

**Review by eljohnson92 — APPROVED** [review]

**Pull request merged**